### ISSUE TITLE
[1.329] Baked GI Issues

### ISSUE THREAD

**Issue by Drexen** [LogicalError/realtime-CSG-for-unity#93]
https://www.youtube.com/watch?v=LTunq-4elgc

````
(texGenFlags[texGenIndex] & TexGenFlags.Discarded) == TexGenFlags.Discarded
````

Lightmaps completely break when you modify some geometry. Also modifying geometry is really slow. Everything in the scene is static using baked lighting for everything.

Also I undo the deletion of the floor brush and it takes a good 5 seconds on a very sparse scene.

**Comment by LogicalError** [comment]
I'll look into it

**Comment by sonygod** [comment]
+1

**Comment by LogicalError** [comment]
not sure yet why the baking stops working, but slowdown is simply because unity is rebuilding everything every time the csg meshes change ... turning off auto lighting + modifying things isn't slow
I'm not entirely sure what, if anything, I could do about that

**Comment by Drexen** [comment]
I do not have auto-build lighting enabled in the example video. (visible on the right at the start)

**Comment by LogicalError** [comment]
I think is partially related to #116 and I think the other part is just .. Unity :-/

The thing is Unity will rebuild the lightmap for the generated mesh once it changes .. so if you modify a brush, the mesh that's generated for that brush will be changed & Unity will rebuild the lightmaps (hopefully only a subset) and I have no control over that.

The garbage you see when modifying the brushes is the existing lightmap being used for a modified mesh that has different uv coordinates for the next lightmap being created by Unity.
Maybe I can just stop that lightmap from being shown *somehow*, but other than that there's nothing I can do about that.

At first I thought the baking stopped working but in that video when you look in the bottom right corner you see that the lighting is being recalculated, and it's just .. slow
(Unity 5.5+ should be a lot faster by the way)

The Undo being slow is a different issue entirely (and there's another ticket for that #29 )

The slowness of modifying geometry while baking is partially my fault (which I will fix today), and partially the lighting being recalculated in the background.

**Comment by LogicalError** [comment]
Some of these issues should be fixed in 1.328!
You can download this version at the update site https://prenominal.nl/updates/

**Comment by LogicalError** [comment]
I made some more changes in 1.329 that should remove the garbage that you see when modifying brushes.
It does need to rebuild the lighting when you change something, but there's no way around that unfortunately. Let me know what you think.
As usual:
You can download this version at the update site https://prenominal.nl/updates/ 
I will close this ticket once it's available from the asset store (tm)
